NeoML/src/TraditionalML/FloatVector.cpp

```diff
@@ -20,52 +20,47 @@ limitations under the License.
 
 namespace NeoML {
 
-CFloatVector::CFloatVector( int size, const CSparseFloatVector& sparseVector )
+CFloatVector::CFloatVectorBody::CFloatVectorBody( int size )
 {
-	NeoAssert( size >= 0 );
-	auto bodyPtr = FINE_DEBUG_NEW CFloatVectorBody( size );
-	const CSparseFloatVectorDesc& desc = sparseVector.GetDesc();
-	int ptrPos = 0;
-	int ptrSize = sparseVector.NumberOfElements();
-
-	float value = 0;
-	for( int i = 0; i < size; i++ ) {
-		if( ptrPos >= ptrSize || i < desc.Indexes[ptrPos] ) {
-			value = 0;
-		} else {
-			value = desc.Values[ptrPos];
-			ptrPos++;
-		}
-		bodyPtr->Values[i] = value;
-	}
-
-	// No elements should stay unprocessed!
-	NeoAssert( ptrPos == ptrSize );
+	Values.SetSize( size );
+	Desc.Values = Values.GetPtr();
+	Desc.Size = size;
+}
 
-	body = bodyPtr;
+CFloatVector::CFloatVector( int size, const CSparseFloatVector& sparseVector ) :
+	CFloatVector( size, sparseVector.GetDesc() )
+{
+	NeoAssert( desc.Indexes != nullptr );
```

Comment:
What is 'desc'?

Reply:
hm.. and how it was compiled?..

Reply:
oh... desc was a read field not in use... fixed.